netutils/interface.py

```diff
@@ -1,8 +1,89 @@
 """Functions for working with interface."""
 
+import itertools
 from .variables import BASE_INTERFACES, REVERSE_MAPPING
 
 
+def interface_tail_addition(interface_number):
+    """Takes the tail end of the interface and sums it up to use for sorting a list of interfaces.
+
+    Args:
+        interface_number (str): Interface number I.E: 1/0/1, 1/0, 0/1
+
+    Returns:
+        int: Sum of number in interface tail with each one getting a priority of magnitude 10 from right to left.
+
+    Example:
+        >>> from netutils.interface import interface_tail_addition
+        >>> interface_tail_addition("1/0/6")
+        106
+        >>> interface_tail_addition("2/3/6")
+        236
```

Comment:
Does this fail in unexpected ways if any of the numbers are non-single-digits? (`0/0/48`, `10/0/16`)

Reply:
It doesn't appear to:
```python
>>> interface_tail_addition("42/233/6123412")
6129942
>>>
```

Reply:
That would seem unexpected to me. :-) Put another way, shouldn't [`0/0/1`, `0/0/48`, `0/1/1`] sort into that given order? But these would evaluate as `1`, `48`, and `11`, resulting in a sorted order of [`0/0/1`, `0/1/1`, `0/0/48`]

Reply:
```python
>>> from netutils.interface import list_alphabetical
>>> ["Fa0/0/1", "Fa0/0/48", "Fa0/1/1"]
['Fa0/0/1', 'Fa0/0/48', 'Fa0/1/1']
>>>```

Reply:
```python
>>> from netutils.interface import list_alphabetical
>>> list_alphabetical(["Fa0/0/1", "Fa0/0/48", "Fa0/1/1"])
['Fa0/0/1', 'Fa0/1/1', 'Fa0/0/48']
```

Reply:
If the goal is to be able to sort interface IDs, that can be accomplished with tuples:
```python
>>> interfaces = [(0,0,1), (0,0,48), (0,1,1)]
>>> sorted(interfaces)
[(0, 0, 1), (0, 0, 48), (0, 1, 1)]
```
What's the advantage of integers?

Reply:
@micahculpepper That is awesome. I didn't even think of that. Thanks for sharing. I'm going to update my code to use that.

Reply:
That is a great one!! A few more use use cases to consider though. 
`["Fa0/0/2", "Fa0/0/1", "Fa0/0/1.100", "Fa0/0/2.200"]`
`["Gi0/1", "Gi0/0/1", "Gi0/0/2" ]`
`["Gi1/0/1", "Te1/0/1", "Gi1/0/2" ]`

I believe these are all real world use cases.

Reply:
```python
sorted(map(
    lambda iface: (
        lambda tup: (tup[0], *re.split(r"[./]", tup[1]))
    )(split_interface(iface)),
    ifaces
))
```
That works per Micah's recommendation, but you lose the context of the separator.  The reason this is a problem is that "Fa0/1" has lower precedence over "Fa0.1".  Domain logic rears its head on this one.

Reply:
Bad habit, here it is in "real" python.  :)
```python
out = []
for iface in ifaces:
    head, tail = split_interface(iface)
    tup = (head, *re.split(r"[./]", tail))
    out.append(tup)

print(list(sorted(out)))
```